### DISCUSSION TITLE
Deploying on Apache server

### DISCUSSION THREAD

**Discussion by nkalinov** [vercel/next.js#24165] [Help] [answered]
I'm having troubles deploying my Next app on an Apache server. 
- `/_next/data` requests hit a 502
- Deep links to subpages also hit 502 (ex. https://qa.mysupermarket.bg/terms)

Main page: https://qa.mysupermarket.bg/ - see 502 on the 2 SSR-ed pages: terms and privacy.

I've already proxied request to the Next server (ran by `next start`) and also managed to "fix" serving of `/_next/static` assets:
```
# bypass ProxyPass for Next static assets
ProxyPass /_next/static "!"
Alias /_next/static /var/www/mm/qa/current/web-main/.next/static
# Next server
ProxyPass / http://localhost:3000
ProxyPassReverse / http://localhost:3000
```

Also, images that I've placed in /public folder and referenced like `<img src="/bg-btn-google-play.png" ...` are looked against the root folder: `https://qa.mysupermarket.bg/bg-btn-google-play.png` I was expecting to see `/public` in the path. What extra config do I need for this?

What am I missing?
Thanks in advance!

**Comment by nkalinov** [comment, marked as answer]
I found the issue which was caused by a missing trailing slash on the ProxyPass directive.
My final vhost config related to Next server is just this:
```
ProxyPass / http://localhost:3000/
ProxyPassReverse / http://localhost:3000/
```

Also notice that I don't have an Alias from _next/ to .next/ folder because I found it more concise to just change the `distDir` to:
```
module.exports = {
  distDir: '_next',
};
```
Probably there is a reason why that's not the default but I'm pretty sure it is a faster option because it skips 1 Apache lookup of an Alias.

**Reply by ShawnCentra** [reply]
Hey @nkalinov  

I'm currently in the processing of trying to set up my Next.js app on Apache2. 

I've downloaded the project files and ran a build of the project, installed node and PM2. 

Did you follow any instructions/guides on how to go from here?

I managed to reverse proxy on ISS so I believe its a similar process but unsure how to go about it in Apache.

**Reply by MarwenLabidi** [reply]
hello im trying todo the same as you 
do you have any resourse could hope me to achieve that ?
